lang/test/org/partiql/lang/eval/EvaluatingCompilerGroupByTest.kt

```diff
@@ -932,4 +953,96 @@ EvaluatingCompilerGroupByTest : EvaluatorTestBase() {
+            "SELECT MAX(@v2), @v2 FROM `[1, 2.0, 3e0, 4, 5d0]` AS v2",
+            session,
+            ErrorCode.EVALUATOR_VARIABLE_NOT_INCLUDED_IN_GROUP_BY,
+            mapOf(
+                    Property.LINE_NUMBER to 1L,
+                    Property.COLUMN_NUMBER to 19L,
+                    Property.BINDING_NAME to "v2"
+            ),
+            null)
+    }
+
+    @Test
+    fun missingGroupBySFWTest() {
+        checkInputThrowingEvaluationException(
+            """
+            SELECT x.title, MIN(money) as price
+            FROM books AS x, x.price AS money
+            WHERE author like '%Milan%'
+            """.trimIndent(),
+            session,
+            ErrorCode.EVALUATOR_VARIABLE_NOT_INCLUDED_IN_GROUP_BY,
+            mapOf(
+                    Property.LINE_NUMBER to 1L,
+                    Property.COLUMN_NUMBER to 8L,
+                    Property.BINDING_NAME to "x"
+            ),
+            null)
+    }
+
+    @Test
+    fun missingGroupByJoinTest() {
+        checkInputThrowingEvaluationException(
+            """
+            SELECT MAX(o.cost), c.firstName 
+            FROM customers AS c 
+            INNER JOIN orders AS o ON c.customerId = o.customerId
+            """.trimIndent(),
+            session,
+            ErrorCode.EVALUATOR_VARIABLE_NOT_INCLUDED_IN_GROUP_BY,
+            mapOf(
+                    Property.LINE_NUMBER to 1L,
+                    Property.COLUMN_NUMBER to 21L,
+                    Property.BINDING_NAME to "c"
+            ),
+            null)
+    }
+
+    @Test
+    fun missingGroupByHavingTest() {
+        checkInputThrowingEvaluationException(
+            """
+            SELECT MAX(o.cost) as total_cost, o.sellerId AS seller_id
+            FROM orders AS o
+            GROUP BY o.customerId 
+            HAVING COUNT(1) > 1
+            """.trimIndent(),
+            session,
+            ErrorCode.EVALUATOR_VARIABLE_NOT_INCLUDED_IN_GROUP_BY,
+            mapOf(
+                    Property.LINE_NUMBER to 1L,
+                    Property.COLUMN_NUMBER to 35L,
+                    Property.BINDING_NAME to "o"
+            ),
+            null)
+    }
+
+    @Test
+    fun missingGroupByOuterQueryTest() {
+        checkInputThrowingEvaluationException(
+            """
+            SELECT AVG(o.cost) AS avg_cost, o.customerId AS customer_id
+            FROM orders AS o
+            WHERE
+	            o.customerId IN(
+                    SELECT VALUE o.customerId 
+                    FROM orders AS o 
+                    WHERE o.customerId > 123 
+                    GROUP BY o.customerId
+            )
+            """.trimIndent(),
+            session,
+            ErrorCode.EVALUATOR_VARIABLE_NOT_INCLUDED_IN_GROUP_BY,
+            mapOf(
+                    Property.LINE_NUMBER to 1L,
+                    Property.COLUMN_NUMBER to 33L,
+                    Property.BINDING_NAME to "o"
+            ),
+            null)
+    }
 }
```

Comment:
Need to include tests with varying cases. i.e. `SELECT F.name, MAX(f.id) FROM foo AS f` -- for instance, this should still throw the `EVALUATOR_VARIABLE_NOT_INCLUDED_IN_GROUP_BY` error.  

Also add a test that proves `EVALUATOR_BINDING_DOES_NOT_EXIST` is not thrown for `SELECT "F".name, MAX(f.id) FROM foo AS f`.

Reply:
Added some tests just for case-(in)sensitivity. Let me know if they aren't sufficient